### ISSUE TITLE
ui.run(native=True) does not exit python execution when exiting native app.

### ISSUE THREAD

**Issue by henrytheodoroff** [zauberzeug/nicegui#3931]
### Description

```python
from nicegui import ui

ui.label('page with custom title')

ui.run(title='My App', native=True)
print("Done!")
```

When running, I get this output while the native app is open (not on a browser):
![image](https://github.com/user-attachments/assets/842971be-2b34-47a3-8e57-b7ae1918f00b)

Then, after pressing the close button on the app, the console still looks the same as above, and python is still running. 

I expect the print("Done!") to not execute until after the app is closed, and then I expect python to halt execution, but it doesn't. Can someone please point me to how python ends on exit of native app? Thanks in advance.

I am running python 3.11.6 and nicegui 2.5.0.

**Comment by henrytheodoroff** [comment]
I am also not wanting to ctrl + c in the terminal. I want to be able to run this and not have manual interference to stop python.

Also, why does _"Done!"_ get printed twice?

**Comment by rodja** [comment]
Have you seen https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-code-executed-twice?

**Comment by henrytheodoroff** [comment]
I see that now, but my main question still stands. Do I have to shutdown the server manually?

**Comment by rodja** [comment]
I agree. The program should exit automatically if you close the main window. And, at least on OS X, it does exactly that. And with the following modifications you'll only see a single "Done" printed after the window is closed:

```py
from nicegui import ui

ui.label('page with custom title')

if __name__ in ('__main__',):
    ui.run(reload=False, native=True)
    print("Done!")
```

**Comment by Lexstok** [comment]
You can shutdown the server with a button, if that helps ?  Or just call it as a function when you decide that the program is finished.
```
from nicegui import ui, app
from datetime import datetime

ui.button('shutdown', on_click=app.shutdown)
app.on_shutdown(lambda: print(f'shutdown:  {datetime.now()}'))

ui.run(native=True)
```

**Comment by 07rohanraj** [comment]
I want to close the application when we close the app but it's running in the background in the task manager. Can I use it for this purpose.

**Comment by 07rohanraj** [comment]
Okay, but will trigger this app.shutdown event, in your case it's button in the UI but i want it to trigger when window close X button clicked:
I was trying with signal like SIGTERM, but it's not working.

**Comment by falkoschindler** [comment]
I think the problem, that the app not shutting down then the "X" is pressed in native mode, is already discussed in #2107. Let's continue over there. Maybe someone can check the new uvicorn version mentioned in https://github.com/zauberzeug/nicegui/issues/2107#issuecomment-2134640373.
